src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component;
+
+/**
+ * Interface Factory
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Factory
+{
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *      An Interruptive Modal disrupts the user in critical situation, forcing him or her to focus on the task at hand.
+     *   composition: >
+     *      The Modal states why this situation needs attention and may point out consequences.
+     *   effect: >
+     *      All controls of the original context are inaccessible until the Modal is completed. Upon completion the user returns to the original context.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *        Due to the heavily disruptive nature of this type of Modal it MUST be restricted to critical situations (e.g. loss of data).
+     *     2: >
+     *        All delete-actions are critical situations and MUST be implemented as an Interruptive Modal.
```

Comment:
I agree with Richard. Using (-) on multi value inputs is almost like deleting characters in a text input field. I would prefer not to have modal here. Even if after hitting "Save" both changes/deletions will be reflected on the persistent layer.